owners/src/parser.js

```diff
@@ -167,53 +167,64 @@ class OwnersParser {
    *     rule.
    */
   _parseOwnersDict(ownersPath, ownersDict) {
-    let [[pattern, ownersList]] = Object.entries(ownersDict);
+    const [[fullPattern, ownersList]] = Object.entries(ownersDict);
     const rules = [];
     const errors = [];
+    let patternList = [];
 
-    const owners = [];
-    const isRecursive = pattern.startsWith(GLOB_PATTERN);
-    if (isRecursive) {
-      pattern = pattern.slice(GLOB_PATTERN.length);
-    }
-
-    if (pattern.indexOf('/') !== -1) {
-      errors.push(
-        new OwnersParserError(
-          ownersPath,
-          `Failed to parse rule for pattern '${pattern}'; ` +
-            `directory patterns other than '${GLOB_PATTERN}' not supported`
-        )
-      );
-    } else if (typeof ownersList === 'string') {
-      const lineResult = this._parseOwnersLine(ownersPath, ownersList);
-      owners.push(...lineResult.result);
-      errors.push(...lineResult.errors);
+    // TODO(rcebulko): Remove backwards-compatibility once all owners files have
+    // been updated to use brace syntax.
+    if (fullPattern.indexOf('{') === -1) {
```

Comment:
I believe this supports ES6 syntax. This can be rewritten as `!fullPattern.includes('{')`.